### PR TITLE
Make auths optional in new MR API

### PR DESCRIPTION
**Pull request by milleruntime**

### CODE REVIEW

File: hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java [hunk 1]
```diff
@@ -124,6 +108,15 @@
    * @since 2.0
    */
   interface InputFormatOptions<T> {
+    /**
+     * Sets the {@link Authorizations} used to scan. Must be a subset of the user's authorizations.
+     * By Default, auths are set to {@link Authorizations#EMPTY}
```
Comment (ctubbsii) [inline hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java]: This doesn't match the new client API, which defaults to the maximal set for the user.

Comment (milleruntime) [reply hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java]: Where is this done?

Comment (ctubbsii) [reply hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java]: https://github.com/apache/accumulo/blob/ed505d076071c792ca97441d17ae2ebd00c34a17/core/src/main/java/org/apache/accumulo/core/clientImpl/AccumuloClientImpl.java#L129-L134
and
https://github.com/apache/accumulo/blob/ed505d076071c792ca97441d17ae2ebd00c34a17/core/src/main/java/org/apache/accumulo/core/clientImpl/AccumuloClientImpl.java#L204-L209

**Comment by milleruntime** [comment]
@ctubbsii I made the builder get all of the users's auths in f5308c7

**Pull request merged**